Sephora-M/assignment1/SinglyLinkedListTest.java

```diff
@@ -0,0 +1,49 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.ExpectedException;
+
+class SinglyLinkedListTest {
+
+	
+	@Rule
+    public ExpectedException thrown = ExpectedException.none();
+	
+	@Test
+	void testKToLast() {
+		SinglyLinkedListNode<Integer> one = new SinglyLinkedListNode<Integer>(1);
+		SinglyLinkedListNode<Integer> two = new SinglyLinkedListNode<Integer>(2);
+		SinglyLinkedListNode<Integer> three = new SinglyLinkedListNode<Integer>(3);
+		SinglyLinkedListNode<Integer> four = new SinglyLinkedListNode<Integer>(4);
+		SinglyLinkedListNode<Integer> five = new SinglyLinkedListNode<Integer>(5);
+		SinglyLinkedListNode<Integer> six = new SinglyLinkedListNode<Integer>(6);
+		SinglyLinkedListNode<Integer> seven = new SinglyLinkedListNode<Integer>(7);
+		SinglyLinkedListNode<Integer> eight = new SinglyLinkedListNode<Integer>(8);
+		SinglyLinkedListNode<Integer> nine = new SinglyLinkedListNode<Integer>(9);
+		SinglyLinkedListNode<Integer> ten = new SinglyLinkedListNode<Integer>(10);
+		
+		one.setNext(two);
```

Comment:
Adding new elements like these might not be very scalable.  In the future in a similar situation, it would be nice to add a class named SinglyLinkedList that would hold a pointer to the first and last elements of the list . It would also contain methods like add(SinglyLinkedListNode) and maybe popFront() or popEnd().
The add(SinglyLinkedListNode) method would do what you're doing here. It would find the last element in the list and append the new element to it (modify last element to point to the new one, change the lastElement pointer/reference to the new element). Then you could even do something like:
`for (int i = 0; i < 100; ++i) list.add(new SinglyLinkedListNode<Integer>(i);`

Reply:
Much more elegant indeed, thanks!